spec/interfaces/udp_interface_spec.rb

```diff
@@ -17,7 +17,10 @@ module Cosmos
   describe UdpInterface do
     describe "initialize" do
       it "initializes the instance variables" do
-        i = UdpInterface.new('localhost','8888','8889','8890','127.0.0.1','64','5','5','127.0.0.1')
+        i = UdpInterface.new('localhost','8888','8889','8890','localhost','64','5','5','localhost')
+        expect(i.instance_variable_get("@hostname")).to eql '127.0.0.1'
+        expect(i.instance_variable_get("@interface_address")).to eql '127.0.0.1'
+        expect(i.instance_variable_get("@bind_address")).to eql '127.0.0.1'
```

Comment:
The bug was everything getting converted to 127.0.0.1.  Should add a check for that.

Reply:
I had the bug logic backwards. I updated the test.